pages/treeherder.py

```diff
@@ -64,3 +73,22 @@ def __init__(self, base_url, selenium):
     @property
     def is_job_status_visible(self):
         return self.is_element_visible(self._job_header_locator)
+
+
+class SingleResultPage(Page):
```

Comment:
Only the query string changes in the URL, the two pages most likely share the same view in the application. The log viewer has a very different DOM, and so I wouldn't consider it comparable to the result sets page.

You could have `SingleResultPage` extending `TreeherderPage` and then you'd only need to implement locators and functions that are specific to when there's a single result show in the page. That said, I suspect anything that's relevant to a single result shown is also relevant to multiple results.